perfTest/recreateDb.js

```diff
@@ -0,0 +1,14 @@
+const { Pool } = require("pg");
+const { parse } = require("pg-connection-string");
+
+const config = parse(process.env.PERF_DATABASE_URL);
+// don't connect to the provided db, or we can't drop it
+const pgPool = new Pool({ ...config, database: undefined });
```

Comment:
I think this won't work if a default database isn't available. (e.g. my username is "benjie" and there's no "benjie" database, so I think this would error for me).

I get around this by using a database name that I know almost everyone has: `template1`:

```suggestion
const pgPool = new Pool({ ...config, database: "template1" });
```

Reply:
With this change, I get an error 

> error: source database "template1" is being accessed by other users

With undefined I don't get this. I tried changing the user name to one without a corresponding db though, and got the error as you expected. Any ideas?

Reply:
Could we use `postgres` as the default? From the [docs:](https://www.postgresql.org/docs/current/creating-cluster.html) 

> The database server itself does not require the postgres database to exist, but many external utility programs assume it exists.

Reply:
Hmmm... From what I've experienced both `undefined` and `"postgres"` cannot be relied on. `undefined` will definitely fail on my machine, I very deliberately do not have a default database because I run so many tools it ensures that I remember to connect to the right DB. `template1` is what we use in all the other tools because it's almost guaranteed to exist. Any idea what you have that is connected to the template1 DB and causing this issue? You shouldn't generally have anything connected to it. Try `select * from pg_stat_activity` (https://www.postgresql.org/docs/current/monitoring-stats.html#PG-STAT-ACTIVITY-VIEW) to see what it is.

If you've not factored in the `pgClient` changes suggested below (e.g. you're still using `pgPool.query` in `recreateDb.js`) then this might be the reason - the "race conditions" I mentioned (two clients connected to the same database at the same time, rather than using just one client).

Reply:
(You figured this out below.)